packages/components/src/duotone-picker/types.ts

```diff
@@ -0,0 +1,61 @@
+export type DuotonePickerProps = {
+	/**
+	 * Whether there should be a button to clear the duotone value.
+	 *
+	 * @default true
+	 */
+	clearable?: boolean;
+	/**
+	 * Whether there should be an `unset` option.
+	 *
+	 * @default true
+	 */
+	unsetable?: boolean;
+	/**
+	 * Array of color presets of the form `{ color: '#000000', name: 'Black', slug: 'black' }`.
+	 */
+	colorPalette: Color[];
+	/**
+	 * Array of duotone presets of the form `{ colors: [ '#000000', '#ffffff' ], name: 'Grayscale', slug: 'grayscale' }`.
+	 */
+	duotonePalette: DuotoneColor[];
+	/**
+	 * Whether custom colors should be disabled.
+	 *
+	 * @default false
+	 */
+	disableCustomColors?: boolean;
+	/**
+	 * Whether custom duotone values should be disabled.
+	 *
+	 * @default false
+	 */
+	disableCustomDuotone?: boolean;
+	/**
+	 * An array of colors for the duotone effect.
+	 */
+	value?: string[] | 'unset';
```

Comment:
I considered whether it was worth doing a tuple (`[ string, string ]`) here instead of `string[]`, but decided against it for two reasons:

- The upstream `CustomGradientPicker` accepts `string[]`, so we'll need to do some added type massaging there.
- I noticed when typing the stories that consumers may be forced to do `[ '#foo', '#bar' ] as const` to make the type checks pass, which is non-obvious.

Does that sound reasonable?

Reply:
A tuple would be definitely better in representing the actual type that the component is expecting — without it, TS wouldn't be able to detect a malformed `value` (ie. `[]`, `[ '#fff' ]` etc).

But the points that you make are valid, and therefore I'd be ok with typing it as `string[]`, at least initially. We can always narrow the type (or put more runtime checks in place) later as we see fit

Reply:
As you pointed out recently, the [new const type improvements](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#const-type-parameters) may come handy for this scenario!